File: StockAppApi/processes/discord/src/commandHandler.cpp
#include "commandHandler.h"

#include "boost/format.hpp"

#include "messageParser.h"
#include "logger.h"

namespace DiscordConnector
{
    namespace Src
    {
        CommandHandler::CommandHandler(
            const std::string &token,
            const std::map<std::string, std::string> &webhooks) : tokenM(token),
                                                                  webhooksM(webhooks),
                                                                  discordMessengerM(new Messenger(tokenM)),
                                                                  commmandsM("sendMessage, sendEmbed")
        {
            try
            {
                for (auto const webhook : webhooksM)
                {
                    discordMessengerM->addWebhook(webhook.first, webhook.second);
                }

                // Send a message to general webhook
                discordMessengerM->sendEmbed("general", "server online", "Keep up!");
            }
            catch (const std::exception &e)
            {
                Base::Src::Log::LogCritical(__FILE__, __LINE__, e.what());
                throw;
            }
        }

        Base::Interface::Response CommandHandler::execute(std::string message)
        {
            try
            {
                auto content = Base::Src::parseMessage(message, "--");

                if (content["command"] == "sendMessage")
                {
                    discordMessengerM->sendMessage(content["channel"], content["message"]);
                }
                else if (content["command"] == "sendEmbed")
                {
                    discordMessengerM->sendEmbed(content["channel"], content["title"], content["message"]);
                }

                return Base::Interface::Response{content["message"], Base::Commons::None,
                                                 "", true};
            }
            catch (const std::exception &e)
            {
                Base::Src::Log::LogCritical(__FILE__, __LINE__, e.what());
                return Base::Interface::Response{"exception", Base::Commons::BadRequest,
                                                 e.what(), false};
            }

            return Base::Interface::Response{"NoContent", Base::Commons::NoContent,
                                             "", false};
        }

        std::string CommandHandler::getCommandsAsStr()
        {
            return commmandsM;
        }
    }
}

File: StockAppApi/processes/python/talib/src/command_handler.py
from StockAppApi.base.python.interface.commandHandlerIf import CommandHandlerIf, Response
from StockAppApi.base.python.src.message_parser import parse_message
from StockAppApi.processes.python.talib.src.aggregator import Aggregator

class CommandHandler(CommandHandlerIf):
    def __init__(self, selected_stocks, aggregator: Aggregator) -> None:
        super().__init__()
        self.commands = "select"
        self.selected_stocks = selected_stocks
        self.aggregator = aggregator
        
    def execute(self, message: str) -> Response:
        try:
            result = self.aggregator.get_analysis(message)
                # send for filter
            return Response(result, 200, "", True)
        except Exception as e:
            # Base::Src::Log::LogCritical(__FILE__, __LINE__, e.what());
            return Response("exception", 400, e.args, False)

    def get_command_as_str(self) -> str:
        return self.commands
